internal/objects/uniquehostname.go

```diff
@@ -0,0 +1,197 @@
+	present, routeList := n.HostnameToRoutesStore.Get(hostName)
+	utils.AviLog.Debugf("Details fetched from HostnameToRoutesStore is: Present: %v and routeList: %v", present, utils.Stringify(routeList))
+	if !present {
+		// host not present, add it to activeRoute list
+		// Allocate map
+		localRouteList.activeRoutes = make(map[string]metav1.Time)
+
+		localRouteList.activeRoutes[route.RouteNSRouteName] = route.CreationTime
+		n.HostnameToRoutesStore.AddOrUpdate(hostName, localRouteList)
+		utils.AviLog.Debugf("Active routes after appending: %v", utils.Stringify(localRouteList.activeRoutes))
+		return true, nil, nil
+	}
+	existingRoutes := routeList.(RouteList)
+	// Check: route is presnt in active list or not
+	if _, found := existingRoutes.activeRoutes[route.RouteNSRouteName]; found {
+		utils.AviLog.Debugf("key %s found in active routes.", route.RouteNSRouteName)
+		// ingest only that route
+		return true, nil, nil
+	}
+
+	// Key: not found in active and inactive list, now compare namespace present in activeRoutes
+	sameNamespace := false
+	currentRouteNamespace, _ := ExtractTypeNameNamespace(route.RouteNSRouteName)
+	// TODO: Debug statements can be reduced
+	utils.AviLog.Debugf("Namespace of current route is: %s", currentRouteNamespace)
+
+	// go through each active route and compare namespace with active routes.
+	// Active route should contains routes from same namespace
+	// Need to check in case of delete of hostname, are we deleting all these list and entry from internal datastore.
+	for nsName, creationTime := range existingRoutes.activeRoutes {
+		utils.AviLog.Debugf("NamespaceName from active route list is: %v and creation timestamp: %v", nsName, creationTime)
+		existingActiveRoutesNamespace, _ := ExtractTypeNameNamespace(nsName)
+		utils.AviLog.Debugf("Extracted out namespace is: %s", existingActiveRoutesNamespace)
+		if currentRouteNamespace == existingActiveRoutesNamespace {
+			// add it to active list
+			sameNamespace = true
+			break
+		}
+	}
+	// With new logic, we need to check length of active route list. if it is empty then, we need to take another namespace
+	// from inactive list and fetch all routes of it with given fqdn and ingest it.
+
+	if sameNamespace {
+		utils.AviLog.Debugf("Namespace is same. Attaching Route %v to active route list", route.RouteNSRouteName)
+		existingRoutes.activeRoutes[route.RouteNSRouteName] = route.CreationTime
+		n.HostnameToRoutesStore.AddOrUpdate(hostName, existingRoutes)
+		// DO not pass anything. Just return true so to process given route
+		return true, nil, nil
+	}
+	utils.AviLog.Debugf("Route %v is not accepted.", route.RouteNSRouteName)
+	return false, nil, nil
+}
+
+func ExtractTypeNameNamespace(nsName string) (string, string) {
+	if nsName != "" {
+		// changed from Split to SplitN
+		spliStrings := strings.SplitN(nsName, "/", 3)
+		if len(spliStrings) == 3 {
+			return spliStrings[1], spliStrings[2]
```

Comment:
As per functions nomenclature, we should be returning Type, name, namespace but we are returning only name and namespace here